Calling `gh bbs2gh migrate-repo` without specifying `--github-repo` triggers an error

## Description

If `gh bbs2gh migrate-repo` is called without specifying `--github-repo`, a `null` value is sent in the GraphQL query and a `Variable $repositoryName of type String! was provided invalid value` error is returned by the API.

```
[10:32] [DEBUG] HTTP POST: https://api.github.com/graphql
[10:32] [DEBUG] HTTP BODY: {"query":"\n                mutation startRepositoryMigration(\n                    $sourceId: ID!,\n                    $ownerId: ID!,\n                    $sourceRepositoryUrl: URI!,\n                    $repositoryName: String!,\n                    $continueOnError: Boolean!,\n                    $gitArchiveUrl: String,\n                    $metadataArchiveUrl: String,\n                    $accessToken: String!,\n                    $githubPat: String,\n                    $skipReleases: Boolean,\n                    $lockSource: Boolean) { \n                startRepositoryMigration(\n                    input: { \n                        sourceId: $sourceId,\n                        ownerId: $ownerId,\n                        sourceRepositoryUrl: $sourceRepositoryUrl,\n                        repositoryName: $repositoryName,\n                        continueOnError: $continueOnError,\n                        gitArchiveUrl: $gitArchiveUrl,\n                        metadataArchiveUrl: $metadataArchiveUrl,\n                        accessToken: $accessToken,\n                        githubPat: $githubPat,\n                        skipReleases: $skipReleases,\n                        lockSource: $lockSource\n                    }\n                ) {\n                    repositoryMigration {\n                        id,\n                        migrationSource {\n                            id,\n                            name,\n                            type\n                        },\n                        sourceUrl,\n                        state,\n                        failureReason\n                    }\n                  } }","variables":{"sourceId":"MS_kgDaACRkM2RiNWUyMC1hMzNkLTQ3MDUtYmJkYy0wYjg4MTc2NWVkNDA","ownerId":"O_kgDOBva8Sg","sourceRepositoryUrl":"https://not-used","repositoryName":null,"continueOnError":true,"gitArchiveUrl":"REDACTED","metadataArchiveUrl":"https://not-used","accessToken":"not-used","githubPat":"***","skipReleases":false,"lockSource":false},"operationName":"startRepositoryMigration"}
[10:32] [DEBUG] GITHUB REQUEST ID: DA51:2465:B5530D:B80348:63906BB2
[10:32] [DEBUG] RESPONSE (OK): {"errors":[{"extensions":{"value":null,"problems":[{"path":[],"explanation":"Expected value to not be null"}]},"locations":[{"line":6,"column":21}],"message":"Variable $repositoryName of type String! was provided invalid value"}]}
[10:32] [ERROR] OctoshiftCLI.OctoshiftCliException: Variable $repositoryName of type String! was provided invalid value
   at OctoshiftCLI.GithubApi.EnsureSuccessGraphQLResponse(JObject response)
   at OctoshiftCLI.GithubApi.StartMigration(String migrationSourceId, String sourceRepoUrl, String orgId, String repo, String sourceToken, String targetToken, String gitArchiveUrl, String metadataArchiveUrl, Boolean skipReleases, Boolean lockSource)
   at OctoshiftCLI.GithubApi.StartBbsMigration(String migrationSourceId, String orgId, String repo, String targetToken, String archiveUrl)
   at OctoshiftCLI.BbsToGithub.Handlers.MigrateRepoCommandHandler.ImportArchive(MigrateRepoCommandArgs args, String archiveUrl)
   at OctoshiftCLI.BbsToGithub.Handlers.MigrateRepoCommandHandler.Handle(MigrateRepoCommandArgs args)
   at OctoshiftCLI.Extensions.CommandExtensions.<>c__DisplayClass1_0`3.<<ConfigureCommand>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Invocation.AnonymousCommandHandler.InvokeAsync(InvocationContext )
   at System.CommandLine.Invocation.InvocationPipeline.<>c__DisplayClass4_0.<<BuildInvocationChain>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass17_0.<<UseParseErrorReporting>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass12_0.<<UseHelp>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass22_0.<<UseVersionOption>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass19_0.<<UseTypoCorrections>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<UseSuggestDirective>b__18_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass16_0.<<UseParseDirective>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c.<<RegisterWithDotnetSuggest>b__5_0>d.MoveNext()
--- End of stack trace from previous location ---
   at System.CommandLine.Builder.CommandLineBuilderExtensions.<>c__DisplayClass8_0.<<UseExceptionHandler>b__0>d.MoveNext()
```

We should either validate that the repo name is provided up-front or have a sensible default.

--- a/src/OctoshiftCLI.Tests/bbs2gh/Handlers/MigrateRepoCommandHandlerTests.cs
+++ b/src/OctoshiftCLI.Tests/bbs2gh/Handlers/MigrateRepoCommandHandlerTests.cs
@@ -724,6 +724,7 @@ namespace OctoshiftCLI.Tests.BbsToGithub.Handlers
                 BbsRepo = BBS_REPO,
                 AzureStorageConnectionString = AZURE_STORAGE_CONNECTION_STRING,
                 GithubOrg = GITHUB_ORG,
+                GithubRepo = GITHUB_REPO
             };
             await _handler.Handle(args);
 
@@ -744,6 +745,7 @@ namespace OctoshiftCLI.Tests.BbsToGithub.Handlers
                 ArchivePath = ARCHIVE_PATH,
                 AzureStorageConnectionString = AZURE_STORAGE_CONNECTION_STRING,
                 GithubOrg = GITHUB_ORG,
+                GithubRepo = GITHUB_REPO
             };
             await _handler.Handle(args);
 
@@ -759,7 +761,9 @@ namespace OctoshiftCLI.Tests.BbsToGithub.Handlers
             var args = new MigrateRepoCommandArgs
             {
                 ArchiveUrl = ARCHIVE_URL,
-                AzureStorageConnectionString = AZURE_STORAGE_CONNECTION_STRING
+                AzureStorageConnectionString = AZURE_STORAGE_CONNECTION_STRING,
+                GithubOrg = GITHUB_ORG,
+                GithubRepo = GITHUB_REPO
             };
             await _handler.Handle(args);
 
@@ -1070,6 +1074,67 @@ namespace OctoshiftCLI.Tests.BbsToGithub.Handlers
                 .Should()
                 .ThrowExactlyAsync<OctoshiftCliException>()
                 .WithMessage("*SSH*SMB*--bbs-server-url*");
+        }
+
+        [Fact]
+        public async Task It_Throws_If_Github_Org_Is_Provided_But_Github_Repo_Is_Not()
+        {
+            // Act
+            var args = new MigrateRepoCommandArgs
+            {
+                GithubOrg = GITHUB_ORG,
+                BbsServerUrl = BBS_SERVER_URL,
+                SshUser = SSH_USER,
+                SshPrivateKey = PRIVATE_KEY,
+                BbsProject = BBS_PROJECT,
+                BbsRepo = BBS_REPO,
+                GithubPat = GITHUB_PAT,
+                BbsUsername = BBS_USERNAME,
+                BbsPassword = BBS_PASSWORD,
+                AzureStorageConnectionString = AZURE_STORAGE_CONNECTION_STRING
+            };
+
+            // Assert
+            await _handler.Invoking(x => x.Handle(args))
+                .Should()
+                .ThrowExactlyAsync<OctoshiftCliException>()
+                .WithMessage("*--github-repo*");
+        }
+
+        [Fact]
+        public async Task It_Throws_If_Archive_Url_Is_Provided_But_Github_Org_Is_Not()
+        {
+            // Act
+            var args = new MigrateRepoCommandArgs
+            {
+                GithubPat = GITHUB_PAT,
+                ArchiveUrl = ARCHIVE_URL,
+                GithubRepo = GITHUB_REPO
+            };
+
+            // Assert
+            await _handler.Invoking(x => x.Handle(args))
+                .Should()
+                .ThrowExactlyAsync<OctoshiftCliException>()
+                .WithMessage("*--github-org*");
+        }
+
+        [Fact]
+        public async Task It_Throws_If_Archive_Url_Is_Provided_But_Github_Repo_Is_Not()
+        {
+            // Act
+            var args = new MigrateRepoCommandArgs
+            {
+                GithubPat = GITHUB_PAT,
+                ArchiveUrl = ARCHIVE_URL,
+                GithubOrg = GITHUB_ORG
+            };
+
+            // Assert
+            await _handler.Invoking(x => x.Handle(args))
+                .Should()
+                .ThrowExactlyAsync<OctoshiftCliException>()
+                .WithMessage("*--github-repo*");
         }
     }
 }

--- a/src/bbs2gh/Handlers/MigrateRepoCommandHandler.cs
+++ b/src/bbs2gh/Handlers/MigrateRepoCommandHandler.cs
@@ -130,7 +130,7 @@ public class MigrateRepoCommandHandler : ICommandHandler<MigrateRepoCommandArgs>
 
     private bool ShouldImportArchive(MigrateRepoCommandArgs args)
     {
-        return args.ArchiveUrl.HasValue();
+        return args.ArchiveUrl.HasValue() || args.GithubOrg.HasValue();
     }
 
     private async Task<string> DownloadArchive(long exportId)
@@ -421,6 +421,11 @@ public class MigrateRepoCommandHandler : ICommandHandler<MigrateRepoCommandArgs>
         {
             ValidateUploadOptions(args);
         }
+
+        if (ShouldImportArchive(args))
+        {
+            ValidateImportOptions(args);
+        }
     }
 
     private void ValidateDownloadOptions(MigrateRepoCommandArgs args)
@@ -481,6 +486,19 @@ public class MigrateRepoCommandHandler : ICommandHandler<MigrateRepoCommandArgs>
         else if (args.AwsAccessKey.HasValue() || args.AwsSecretKey.HasValue())
         {
             throw new OctoshiftCliException("--aws-access-key and --aws-secret-key can only be provided with --aws-bucket-name.");
+        }
+    }
+
+    private void ValidateImportOptions(MigrateRepoCommandArgs args)
+    {
+        if (args.GithubOrg.IsNullOrWhiteSpace())
+        {
+            throw new OctoshiftCliException("--github-org must be provided in order to import the Bitbucket archive.");
+        }
+
+        if (args.GithubRepo.IsNullOrWhiteSpace())
+        {
+            throw new OctoshiftCliException("--github-repo must be provided in order to import the Bitbucket archive.");
         }
     }
 

